CallAutomation_OpenAI_Sample/src/main/java/com/communication/callautomation/ProgramSample.java

```diff
@@ -0,0 +1,326 @@
+        }
+        return ResponseEntity.ok().body("");
+    }
+
+    private void handleIncomingCall(final BinaryData eventData) {
+        JSONObject data = new JSONObject(eventData.toString());
+        String callbackUri;
+        AnswerCallOptions options;
+        String cognitiveServicesUrl;
+
+        try {
+            callbackUri = String.format("%s/%s?callerId=%s",
+                    appConfig.getCallBackUri(),
+                    UUID.randomUUID(),
+                    data.getJSONObject("from").getString("rawId"));
+            cognitiveServicesUrl = new URI(appConfig.getCognitiveServicesUrl()).toString();
+            options = new AnswerCallOptions(data.getString(INCOMING_CALL_CONTEXT),
+                    callbackUri).setCognitiveServicesEndpoint(cognitiveServicesUrl);
+            Response<AnswerCallResult> answerCallResponse = client.answerCallWithResponse(options, Context.NONE);
+
+            log.info("Incoming call answered. Cognitive Services Url: {}\nCallbackUri: {}\nCallConnectionId: {}",
+                    cognitiveServicesUrl,
+                    callbackUri,
+                    answerCallResponse.getValue().getCallConnectionProperties().getCallConnectionId());
+        } catch (Exception e) {
+            log.error("Error getting recording location info {} {}",
+                    e.getMessage(),
+                    e.getCause());
+        }
+    }
+
+    private ResponseEntity<SubscriptionValidationResponse> handleSubscriptionValidation(final BinaryData eventData) {
+        try {
+            log.info("Received Subscription Validation Event from Incoming Call API endpoint");
+            SubscriptionValidationEventData subscriptioneventData = eventData.toObject(SubscriptionValidationEventData.class);
+            SubscriptionValidationResponse responseData = new SubscriptionValidationResponse();
+            responseData.setValidationResponse(subscriptioneventData.getValidationCode());
+            return ResponseEntity.ok().body(responseData);
+        } catch (Exception e) {
+            log.error("Error at subscription validation event {} {}",
+                    e.getMessage(),
+                    e.getCause());
+            return ResponseEntity.internalServerError().body(null);
+        }
+    }
+
+    private void handleRecognizeRequest(
+            final String message,
+            final String callConnectionId,
+            final String callerId)
+    {
+        String targetParticipant = callerId.replaceAll("\\s", "+");
+        TextSource textSource = new TextSource()
+                .setText(message)
+                .setVoiceName("en-US-NancyNeural");
+        CallMediaRecognizeSpeechOptions options = new CallMediaRecognizeSpeechOptions(
+                CommunicationIdentifier.fromRawId(targetParticipant),
+                Duration.ofSeconds(5))
+                .setInterruptPrompt(false)
+                .setPlayPrompt(textSource)
+                .setOperationContext("GetFreeFormText")
```

Comment:
NIT: should we use the word Freeform in any of our samples ? I would avoid using it since its internal name and not exposed to customers in any form

Reply:
yes, thanks, I will remove it now